packages/blade/src/components/Button/_decisions/decisions.md

```diff
@@ -0,0 +1,81 @@
+# Button Decisions <!-- omit in toc -->
+- [Design](#design)
+- [Internal BaseButton Component](#internal-basebutton-component)
+- [Button Component](#button-component)
+- [Component Breakdown](#component-breakdown)
+- [Open Questions](#open-questions)
+
+## Design
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=5200%3A0) to all variants of the Button component
+
+## Internal BaseButton Component
+-  Internal component that exposes certain extra props to enable creation of Component Level Action Buttons like `positive`, `negative`, `neutral`
+-  **BaseButton API**:
+   -  **`variant`**: `primary` | `secondary` | `tertiary` | `link`
+      -  **default**: `primary`
+      -  Link will look like an inline Link component but will be rendered as `button` element
+   -  **`intent`**: `positive` | `negative` | `notice` | `info` | `neutral`
+      -  **default**: `undefined`
+      -  We will use this to set the `intent` for internal Blade components that have positive, negative, notice, info, or neutral buttons like an `Alert` component
+   -  **`contrast`**: `high` | `low`
+      -  **default**: `low`
+      -  We need this for internal Blade components since a `positive` intent could have a `high` as well as a `low` contrast button
+   -  **`size`**: `large` | `medium` | `small` | `xsmall`
+      - **default**: `medium`
+   -  **`children`**: `string`
+   -  **`icon`**: `IconComponent`
```

Comment:
is this going to be a component or just the name of the Icon?
I think the component should be used inside the `BaseButton` and let's only expose the name of the icon that needs to be displayed .. what do you think?

Reply:
It will be the React component element itself. 

Will probably look something like this

```ts
<Button icon={SearchIcon} />
```

Reply:
> let's only expose the name, typography, etc of the icon that needs to be displayed .. what do you think?

So exposing only the name as issues. [Icons won't be treeshakable](https://github.com/razorpay/blade/issues/363)

Reply:
this would mean if consumer won't be able to pass any custom prop to the icon. Is that ok?

Reply:
@aditya-razorpay that's the intention. We wouldn't want someone to use a `xlarge` icon with a `small` button. Do you forsee any alarming use-cases where we might need to pass some custom props to the icon directly?

Reply:
the only thing I'm worried about is that as the Icon component api widens, the Button api would also widen. If Button took an iconElement then Button and Icon would still be composable without Button having to extend Icon api.

However I do see your point that we want to place some restrictions on what props are passed to Icon when used inside Button which can't be done without React.cloneElement (and not a big fan of that)

Reply:
I don't see the icon api getting more complex than what it is today except maybe supporting multicolor icon at some point in the future. I think adding another prop to button would still seem better than to open up the icon prop and let the consumer configure it since that would lead to (allow) misuse from day 1 😓